frugen.c

```diff
@@ -460,6 +470,61 @@ int main(int argc, char *argv[])
 					fatal("JSON support was disabled at compile time");
 #endif
 				}
+				else if (use_binary) {
+					int fd = open(optarg, O_RDONLY);
+					if (fd < 0) {
+						fatal("Failed to open file: %s", strerror(errno));
+					}
+
+					fru_t *raw_fru = read_fru_header(fd);
+					if (!raw_fru)
+						fatal("Failed to read fru header");
+
+					if (raw_fru->chassis != 0) {
+						if (lseek(fd, 8 * raw_fru->chassis, SEEK_SET) < 0)
+							fatal("Failed to seek");
+
+						fru_chassis_area_t *chassis_raw =
+							read_fru_chassis_area(fd);
+						bool success = fru_decode_chassis_info(
+							chassis_raw, &chassis);
+						if (!success)
+							fatal("Failed to decode chassis");
+
+						free(chassis_raw);
+						has_chassis = true;
+					}
+					if (raw_fru->board != 0) {
+						if(lseek(fd, 8 * raw_fru->board, SEEK_SET) < 0)
+							fatal("Failed to seek");
+
+						fru_board_area_t *board_raw = read_fru_board_area(fd);
+						bool success = fru_decode_board_info(board_raw, &board);
+						if (!success)
+							fatal("Failed to decode board");
+
+						free(board_raw);
+						has_board = true;
+						has_bdate = true;
+					}
+					if (raw_fru->product != 0) {
+						if (lseek(fd, 8 * raw_fru->product, SEEK_SET) < 0)
+							fatal("Failed to seek");
+
+						fru_product_area_t *product_raw =
+							read_fru_product_area(fd);
+						bool success =
+							fru_decode_product_info(product_raw, &product);
+						if (!success)
+							fatal("Failed to decode product");
+
+						free(product_raw);
+						has_product = true;
+					}
```

Comment:
Looks like you're losing the "internal use" and "multi-record" areas here. The output file won't have them even if the original raw file did. That may defeat the purpose of your change as I understand it.

Reply:
we don't need those right now. If it's alright, I'd like to omit support for multirec and internal in the output for now. The internal area is pretty straightforward, but the multirec area seems to require some parsing, which I'd prefer to not have to implement for this PR.

Reply:
That's ok because libfru initially didn't support creating those areas, but please specify in the help that users may lose their pre-existing areas when using `--raw`, and also please update the README.md with the information about the new feature and its limitations.